File: 2015/day02-2.py
f = open("day02-puzzle-input.txt", "r")

total = 0

for ligne in f:
    split = ligne.split('x')
    l = int(split[0])
    w = int(split[1])
    h = int(split[2])

    trier = sorted([l, w, h])
    print(trier)

    total += l*w*h + 2*trier[0] + 2*trier[1]

print(total)

File: 2021/day02-2.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	contenu, _ := os.ReadFile("day02-puzzle-input.txt")

	tableau := strings.Split(string(contenu), "\n")
	horizontal := 0
	descente := 0
	aim := 0

	for i := range tableau {
		commande := strings.Split(tableau[i], " ")
		valeur, _ := strconv.Atoi(commande[1])
		switch commande[0] {
		case "up":
			aim -= valeur
		case "down":
			aim += valeur
		case "forward":
			horizontal += valeur
			descente += aim * valeur
		}
	}

	fmt.Println(horizontal * descente)
}
